### PR TITLE
feat: auth, authz, bank e2e tests for fiattokenfactory

### PR DESCRIPTION
**Pull request by boojamya**
Includes:
- `TestFiatTFAuth`
- `TestFiatTFAuthzGrant`
- `TestFiatTFAuthzSend`
- `TestFiatTFBankSend`

### CODE REVIEW

File: e2e/fiat_tf_test.go [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+	sendAmount := 1
+	uusdcFee := 5
+	_, err = val.ExecTx(ctx, alice.KeyName(), "bank", "send", alice.KeyName(), bob.FormattedAddress(), fmt.Sprintf("%dustake", sendAmount), "--fees", fmt.Sprintf("%duusdc", uusdcFee))
+	require.ErrorContains(t, err, "the chain is paused")
+
+	bal, err := noble.GetBalance(ctx, alice.FormattedAddress(), "ustake")
+	require.NoError(t, err)
+	require.Equal(t, originalAmount, bal)
+
```
Comment (circle-smartin) [inline e2e/fiat_tf_test.go:49]: What's the expected behaviour for the fees here? Are fees still charged (and so alice's USDC balance drops), or is the transaction dropped before fees are charged so the balance remains? Is this worth adding an assertion on?

File: e2e/fiat_tf_test.go [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, alice)
+
+	_, err = val.ExecTx(ctx, alice.KeyName(), "bank", "send", alice.KeyName(), bob.FormattedAddress(), fmt.Sprintf("%dustake", sendAmount), "--fees", fmt.Sprintf("%duusdc", uusdcFee))
+	require.ErrorContains(t, err, fmt.Sprintf("an address (%s) is blacklisted and can not send tokens: unauthorized", alice.FormattedAddress()))
+
+	bal, err = noble.GetBalance(ctx, alice.FormattedAddress(), "ustake")
+	require.NoError(t, err)
+	require.Equal(t, originalAmount, bal)
+
```
Comment (circle-smartin) [inline e2e/fiat_tf_test.go:63]: same question about checking the usdc balance

File: e2e/fiat_tf_test.go [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, grantee)
+
+	// ACTION: Execute an authz SEND using a TF token from a granter who is blacklisted
+	// EXPECTED: Request fails; Granter is blacklisted
+	// Status:
+	// 	Granter1 has authorized Grantee1 to send 100usdc from their wallet
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, granter)
+
+	preSendBal := bal
```
Comment (circle-smartin) [inline e2e/fiat_tf_test.go:197]: Given that balance stays zero for this entire test (all requests fail), do we need this `preSendBal` variable? Can we always check that it is zero instead? Or is there a reason you prefer to do the checks in this way?

File: e2e/fiat_tf_test.go [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+	preSendBal = bal
+
+	_, err = val.AuthzExec(ctx, grantee, nestedCmd)
+	require.ErrorContains(t, err, "the chain is paused")
+
+	bal, err = noble.GetBalance(ctx, receiver.FormattedAddress(), "uusdc")
+	require.NoError(t, err)
+	// bal should not change
+	require.Equal(t, preSendBal, bal)
+}
```
Comment (circle-smartin) [inline e2e/fiat_tf_test.go:245]: Is there any value in adding a happy path test here (no accounts blacklisted, chain not paused)?

File: e2e/fiat_tf_test.go (hunk 1 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:53]: Good call. The fee is not taken. 
Added a check in: https://github.com/noble-assets/noble/pull/373/commits/93925c17bb45c43e80cdce2516044b89fc9126ab

File: e2e/fiat_tf_test.go (hunk 2 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:67]: Added check in: https://github.com/noble-assets/noble/pull/373/commits/93925c17bb45c43e80cdce2516044b89fc9126ab

File: e2e/fiat_tf_test.go (hunk 3 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go]: Yes, good call. Simplified: 
[`133af8f` (#373)](https://github.com/noble-assets/noble/pull/373/commits/133af8f19831312ba7545a8a7270db1740c66e5e)

File: e2e/fiat_tf_test.go (hunk 4 above)
Comment (boojamya) [reply e2e/fiat_tf_test.go:258]: I think so, mainly to validate the test. Added:
[`b345e59` (#373)](https://github.com/noble-assets/noble/pull/373/commits/b345e59ae591f5c7add6576d727a19e7c51c2c1a)

**Review by kaliweh-circle — APPROVED** [review]

**Review by johnletey — APPROVED** [review]
LGTM! Passing on Cosmos SDK v0.50 ⚡

- ✅ Green CI for `TestFiatTFAuth` ([link](https://github.com/noble-assets/noble/actions/runs/11382658386/job/31666761645?pr=373))
- ✅ Green CI for `TestFiatTFAuthzGrant` ([link](https://github.com/noble-assets/noble/actions/runs/11382658386/job/31666762258?pr=373))
- ✅ Green CI for `TestFiatTFAuthzSend` ([link](https://github.com/noble-assets/noble/actions/runs/11382658386/job/31666762764?pr=373))
- ✅ Green CI for `TestFiatTFBankSend` ([link](https://github.com/noble-assets/noble/actions/runs/11382658386/job/31666763168?pr=373))

NOTE: An expected error message had to be adjusted slightly due to the most recent code changes to the FiatTokenFactory send restriction (https://github.com/noble-assets/noble/pull/373/commits/fc8a898cf86d4f359f388b92628448b6de2c1618). This is due to the error message incorrectly using the `toAddr` instead of the `grantee` [here](https://github.com/circlefin/noble-fiattokenfactory/blob/c20bd0c8442fa9b70dc88d8648a5e337c66d4644/x/fiattokenfactory/keeper/keeper.go#L91). cc @circle-smartin

**Pull request merged**